Scripts/Tracking/Collision/List/SliceCollisions.cs

```diff
@@ -0,0 +1,102 @@
+{
+    using UnityEngine;
+    using System.Linq;
+
+    /// <summary>
+    /// Slices a selection of the collection from the given <see cref="startIndex"/> for the given <see cref="length"/> and provides the sliced collection and the remaining collection separately.
+    /// </summary>
+    public class SliceCollisions : MonoBehaviour
+    {
+        /// <summary>
+        /// The zero-based index which to start the slice at.
+        /// </summary>
+        [Tooltip("The zero-based index which to start the slice at.")]
+        public uint startIndex = 0;
+        /// <summary>
+        /// The number of elements in the to slice.
+        /// </summary>
+        [Tooltip("The number of elements in the to slice.")]
+        public uint length = 1;
+
+        /// <summary>
+        /// Emitted when the sliced elements are taken from the collection.
+        /// </summary>
+        public ActiveCollisions.UnityEvent Sliced = new ActiveCollisions.UnityEvent();
+        /// <summary>
+        /// Emitted when the remaining elements are left after slicing.
+        /// </summary>
+        public ActiveCollisions.UnityEvent Remained = new ActiveCollisions.UnityEvent();
+
+        protected ActiveCollisions.EventData slicedList = new ActiveCollisions.EventData();
+        protected ActiveCollisions.EventData remainingList = new ActiveCollisions.EventData();
+
+        /// <summary>
+        /// Slices the collision collection.
+        /// </summary>
+        /// <param name="originalList">The original collision collection.</param>
+        public virtual void DoSlice(ActiveCollisions.EventData originalList)
+        {
+            Slice(originalList);
+        }
+
+        /// <summary>
+        /// Slices the collision collection and returns a new collection containing the sliced collection.
+        /// </summary>
+        /// <param name="originalList">The original collision collection.</param>
+        /// <returns>The sliced collection.</returns>
+        public virtual ActiveCollisions.EventData Slice(ActiveCollisions.EventData originalList)
+        {
+            slicedList.Clear();
+            remainingList.Clear();
+
+            if (!isActiveAndEnabled)
+            {
+                return originalList;
+            }
+
+            uint actualLength = GetRangeLength(startIndex, length, (uint)originalList.activeCollisions.Count);
+            slicedList.activeCollisions.AddRange(originalList.activeCollisions.GetRange((int)startIndex, (int)actualLength));
+            Sliced?.Invoke(slicedList);
+
+            remainingList.activeCollisions.AddRange(originalList.activeCollisions.Except(slicedList.activeCollisions));
+
+            Remained?.Invoke(remainingList);
+
+            return slicedList;
+        }
+
+        /// <summary>
+        /// Slices the collision collection and returns a new collection containing the sliced collection and provides the remaining collection as an out parameter.

```

Comment:
Let's change this to
>Slices the collision collection

as we don't need to explain what it's returning - that is handled by the other XML tags.